pkg/cmd/audit/aggregate.go

```diff
@@ -0,0 +1,47 @@
+package audit
+
+import (
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	auditv1 "k8s.io/apiserver/pkg/apis/audit/v1"
+	"k8s.io/apiserver/pkg/audit"
+)
+
+func GroupBy(paths []string,  events []*auditv1.Event) map[string][]*auditv1.Event {
```

Comment:
you can group by `user name` passing `[]string{"user.username"`
or by `verb, uri and user name` passing `[]string{"verb", "requestURI", "user.username"}`